src/main/java/catan/services/util/play/CardUtil.java

```diff
@@ -69,6 +71,24 @@ public Integer takeResourceFromRivalsAndGiveItAwayToPlayer(GameUserBean currentG
         return takenResourcesCount;
     }
 
+    public Integer defineRoadsQuantityToBuild(GameUserBean gameUser, GameBean game) throws PlayException {
+        List<EdgeBean> availableEdges = new ArrayList<EdgeBean>(game.fetchEdgesAccessibleForBuildingRoad(gameUser));
+        switch (availableEdges.size()) {
```

Comment:
I would simplify this logic to use somehow pretty if-else statment rather than switch-case